pymc3/tests/test_distributions_random.py

```diff
@@ -688,3 +689,35 @@ def ref_rand(size, w, mu, sd):
                      'sd': Domain([[1.5, 2., 3.]], edges=(None, None))}, 
                      size=1000,
                      ref_rand=ref_rand)
+=======
+    def test_density_dist(self):
+        def ref_rand(size, mu, sd):
+            return st.norm.rvs(size=size, loc=mu, scale=sd)
+        
+        class TestDensityDist(pm.DensityDist):
+
+            def __init__(self, **kwargs):
+                norm_dist = pm.Normal.dist()
+                super(TestDensityDist, self).__init__(logp=norm_dist.logp, random=norm_dist.random)
+
+        pymc3_random(TestDensityDist, {},ref_rand=ref_rand)
+
+        def check_model_samplability(self):
+            model = pm.Model()
+            with model:
+                normal_dist = pm.Normal.dist()
+                density_dist = pm.DensityDist('density_dist', normal_dist.logp, random=normal_dist.random)
+                step = pm.Metropolis()
+                trace = pm.sample(5000, step)
+
+            try:
+                ppc = pm.sample_ppc(trace, samples=500, model=model, size=100)
+                if len(ppc) > 0:
+                    pass
+                else:
+                    return "length of ppc sample is zero"
```

Comment:
Instead of returning we want to something like here: https://github.com/Vaibhavdixit02/pymc3/blob/216da86a6c8f9ae6516d133bc198ac45d678e716/pymc3/tests/test_distributions_random.py#L73 `npt.assert_true(len(ppc) > 0, 'length of ppc sample is zero')`. Check the `np.testing` submodule and other parts of the code for various usage patterns.